File: daily/python/Nov-28-Number-of-ways-to-divide-a-long-corridor.py
class Solution:
    def numberOfWays(self, corridor):
        # numSeats = 0
        # numPlants = 0
        # dividers = 1
        # mod = int(1e9 + 7)

        # for i in corridor:
        #     if i == 'S':
        #         numSeats += 1
        #     if numSeats == 2 and i == 'P':
        #         numPlants += 1
        #     if numSeats == 3:
        #         dividers *= (numPlants + 1)
        #         dividers %= mod
        #         numSeats = 1
        #         numPlants = 0

        # if numSeats < 2: return 0
        # return dividers

        seat = 0
        prev = 0
        res = 1
        for x in range(len(corridor)):
            if (corridor[x] == 'S'):
                if (seat < 2):
                    seat += 1
                else:
                    seat = 1
                    res *= (x - prev)
                prev = x
        return 0 if seat < 2 else res % int(1e9+7)

File: daily/cpp/Nov-28-Number-of-ways-to-divide-a-long-corridor.cpp
#include <string>

using namespace std;
class Solution {
 public:
  int numberOfWays(string corridor) {
    int chairs = 0;
    long long res = 1;

    for (int i = 0; i < corridor.size(); i++) {
      if (corridor[i] == 'S') {
        chairs++;

        while (++i < corridor.size() && corridor[i] != 'S')
          ;
        // the while increment i and check each character in the corridor
        // string. the loop continues until the end of the string or until the
        // next chair.
        if (i < corridor.size()) {
          chairs++;
        }

        int divisions = 1;
        while (++i < corridor.size() && corridor[i] != 'S') {
          divisions++;
        }

        if (divisions > 1 && i < corridor.size()) {
          res = (res * divisions) % static_cast<int>(1e9 + 7);
        }
        i--;
      }
    }

    return (chairs != 0 && chairs % 2 == 0) ? static_cast<int>(res) : 0;
  }
};
